docs/examples/example.ipynb

```diff
@@ -898,6 +899,200 @@
+    "from viam.app.client import AppClient\n",
+    "\n",
+    "async def connect() -> AppClient:\n",
+    "    dial_options = DialOptions(\n",
+    "        auth_entity='<ADDRESS>',\n",
+    "        credentials=Credentials(\n",
+    "            type='robot-location-secret',\n",
+    "            payload='<SECRET>'\n",
+    "        )\n",
+    "    )\n",
+    "    return await AppClient.create(dial_options)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once you have a connected `AppClient`, you can then obtain a `DataClient` as a property."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "metadata": {
+    "tags": [
+     "remove-input"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# Hidden.\n",
+    "from typing_extensions import Self\n",
+    "\n",
+    "from viam.app.data.client import DataClient\n",
+    "from viam.rpc.dial import _dial_direct\n",
+    "\n",
+    "class MockAppClient:\n",
+    "\n",
+    "    @classmethod\n",
+    "    async def create(cls) -> Self:\n",
+    "        self = cls()\n",
+    "        self._channel = await _dial_direct(address=\"localhost:9092\", options=DialOptions(insecure=True))\n",
+    "        self.data_client = DataClient(channel=channel, metadata=None)\n",
+    "        return self\n",
+    "\n",
+    "    def close(self):\n",
+    "        self._channel.close()\n",
+    "\n",
+    "app_client = await MockAppClient.create()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 31,
+   "metadata": {
+    "tags": [
+     "hide-output"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "data_client = app_client.data_client"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This `DataClient` can be used to make method calls that retrieve data from app."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 32,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[fields {\n",
+      "  key: \"PowerPct\"\n",
+      "  value {\n",
+      "    number_value: 0\n",
+      "  }\n",
+      "}\n",
+      "fields {\n",
+      "  key: \"IsPowerred\"\n",
+      "  value {\n",
+      "    bool_value: false\n",
+      "  }\n",
+      "}\n",
+      ", fields {\n",
+      "  key: \"PowerPct\"\n",
+      "  value {\n",
+      "    number_value: 0\n",
+      "  }\n",
+      "}\n",
+      "fields {\n",
+      "  key: \"IsPowered\"\n",
+      "  value {\n",
+      "    bool_value: false\n",
+      "  }\n",
+      "}\n",
+      ", fields {\n",
+      "  key: \"Position\"\n",
+      "  value {\n",
+      "    number_value: 0\n",
+      "  }\n",
+      "}\n",
+      "]\n"
+     ]
+    }
+   ],
+   "source": [
+    "from viam.proto.app.data import Filter\n",
+    "\n",
+    "data = await data_client.tabular_data_by_filter(filter=Filter(robot_name=\"arduino\"))\n",
```

Comment:
Can we flesh out this `Filter` some more? Would love to show how all fields are set, or at a minimum the non-obvious fields (timestamps, tags).